src/main/java/org/csanchez/jenkins/plugins/kubernetes/ProvisioningCallback.java

```diff
@@ -274,15 +274,8 @@ private Container createContainer(KubernetesSlave slave, ContainerTemplate conta
         env.put("JENKINS_SECRET", slave.getComputer().getJnlpMac());
         env.put("JENKINS_NAME", slave.getComputer().getName());
 
-        JenkinsLocationConfiguration locationConfiguration = JenkinsLocationConfiguration.get();
-        String locationConfigurationUrl = locationConfiguration != null ? locationConfiguration.getUrl() : null;
-        String url = StringUtils.isBlank(cloud.getJenkinsUrl()) ? locationConfigurationUrl : cloud.getJenkinsUrl();
+        String url = cloud.getJenkinsUrlOrDie();
 
-        if (url == null) {
-            throw new IllegalStateException("Jenkins URL is null while computing JNLP url");
-        }
-
-        env.put("JENKINS_LOCATION_URL", locationConfigurationUrl);
```

Comment:
As far as I could see this was introduced in https://github.com/jenkinsci/kubernetes-plugin/commit/767240441f93428e914898a707810c44e786afd4 but it is not used anymore.